python/tvm/script/utils.py

```diff
@@ -57,3 +107,32 @@ def from_synr_span(span):
         span.start_column,
         span.end_column,
     )
+
+
+def synr_span_from_tvm(span: Span) -> synr.ast.Span:
+    """Convert a TVM span to a synr span"""
+    return synr.ast.Span(
+        span.source_name.name,
+        span.line,
+        span.column,
+        span.end_line,
+        span.end_column,
+    )
+
+
+def call_with_error_reporting(
+    report_error,
+    node_span,
+    func,
+    *args,
+    **kwargs,
+):
+    """Call function with exception handling and report error using node_span"""
+    try:
+        return func(*args, **kwargs)
+    except DiagnosticError as err:
+        raise err
```

Comment:
```suggestion
    except DiagnosticError:
        raise
```

This gives a cleaner stack trace.